agent/api.go

```diff
@@ -50,6 +54,57 @@ func healthz(connected func() bool) http.HandlerFunc {
 	}
 }
 
+func (a *Agent) handleAPIQueue(ctx context.Context) {
+	ch := make(chan *lasr.Message, 1)
+	go func() {
+		limit := a.config.EventsAPIRateLimit
+		if limit == 0 {
+			limit = rate.Limit(math.Inf(1))
+		}
+		limiter := rate.NewLimiter(limit, a.config.EventsAPIBurstLimit)
+		for {
+			if err := limiter.Wait(ctx); err != nil {
+				// context canceled
+				return
+			}
+			message, err := a.apiQueue.Receive(ctx)
+			if err != nil {
+				if ctx.Err() != nil {
+					close(ch)
+					return
+				}
+				logger.WithError(err).Error("error receiving message from queue")
+				continue
+			}
+			ch <- message
+		}
+	}()
+	for {
+		select {
+		case message, ok := <-ch:
+			if !ok {
+				return
+			}
+			msg := &transport.Message{
+				Type:    transport.MessageTypeEvent,
+				Payload: decompressMessage(message.Body),
+				SendCallback: func(err error) {
+					if err != nil {
+						logger.WithError(err).Error("couldn't send queued message, retrying")
+						_ = message.Nack(true)
```

Comment:
I'm not sure what this function returns without looking at the dependency, but is it anything that would be useful to log?

Reply:
If there is an error writing to disk, the message will be left in the unacked state, and the error will be returned. It will eventually make its way back into the ready state on subsequent queue operations.

There's not really much point in logging the error as subsequent queue operations will just return the same error if there is a problem with the disk.